src/web/app/components/question-responses/grouped-responses/grouped-responses.component.ts

```diff
@@ -42,11 +42,24 @@ export class GroupedResponsesComponent extends InstructorResponsesViewBase imple
     createdAtTimestamp: 0,
   };
 
+  hasNonMissingResponses: boolean = false;
+
   constructor() {
     super();
   }
 
   ngOnInit(): void {
+    for (const question of this.responses) {
```

Comment:
- `this.hasNonMissingResponses ` -> `this.hasRealResponses` (or we can just go opposite `this.areAllMissingResponses`). Reasons: The name `hasNonMissingResponses` seems it requires non missing response while it is not true in the case.

- And to save code :P
```javascript
this.hasNonMissingResponses = this.responses.some((question) => question.allResponses.some((response) => !response.isMissingResponse))
```